include/ddsrouter/core/DDSRouter.hpp

```diff
@@ -43,73 +43,206 @@ class DDSRouter
+     */
     DDSRouter(
             const Configuration& configuration);
 
+    /**
+     * @brief Destroy the DDSRouter object
+     *
+     * Stop the DDSRouter
+     * Destroy all Bridges
+     * Destroy all Participants
+     */
     virtual ~DDSRouter();
 
     // EVENTS
-    void reload_configuration(
+    /**
+     * @brief Reload the allowed topic configuration
+     *
+     * @warning Unsupported
+     * @todo Add noexcept when method ready
+     *
+     * @param [in] configuration : new configuration
+     *
+     * @return \c RETCODE_OK if configurations has been updated correctly
+     * @return \c RETCODE_NO_DATA if new configuration has not changed
+     * @return \c RETCODE_BAD_PARAMETER if configurations is bad formed
+     * @return \c RETCODE_ERROR if any other erro has ocurred
+     */
+    ReturnCode reload_configuration(
             const Configuration& configuration);
 
-    void stop();
+    /**
+     * @brief Start communication in DDS Router
+     *
+     * Enable every topic Bridge.
+     *
+     * @note this method returns a ReturnCode for future possible errors
+     *
+     * @return \c RETCODE_OK always
+     */
+    ReturnCode start() noexcept;
+
+    /**
+     * @brief Stop communication in DDS Router
+     *
+     * Disable every topic Bridge.
+     *
+     * @note this method returns a ReturnCode for future possible errors
+     *
+     * @return \c RETCODE_OK always
+     */
+    ReturnCode stop() noexcept;
 
 protected:
 
     /////
-    // INTERNAL METHODS
+    // INTERNAL INITIALIZATION METHODS
 
-    //! Load allowed topics from configuration
+    /**
+     * @brief Load allowed topics from configuration
+     *
+     * @throw \c ConfigurationException in case the yaml inside allowedlist is bad formed
+     */
     void init_allowed_topics_();
 
-    //! Create participants and add it to the participants database
+    /**
+     * @brief  Create participants and add it to the participants database
+     *
+     * @throw \c InitializationException in case \c IParticipants creation fails.
+     */
     void init_participants_();
 
-    //! Create bridges from topics
+    /**
+     * @brief  Create disabled bridges for every real topic
+     *
+     * @throw \c InitializationException in case \c IWriters or \c IReaders creation fails.
+     */
     void init_bridges_();
 
-    //! New Topic found, check if it should be activated
+    /////
+    // INTERNAL AUXILIAR METHODS
+
+    /**
+     * @brief Method called every time a new discovery endpoint has been discovered/updated
+     *
+     * This method is called with the topic of a new/updated \c Endpoint discovered.
+     * If the DDSRouter is enabled, the new Bridge is created and enabled.
+     *
+     * @note This is the only method that adds topics to \c current_topics_
+     *
+     * @param [in] topic : topic discovered
+     */
     void discovered_topic_(
```

Comment:
`@throw` missing?

Reply:
No, this method does not throw any exception (that's why `noexcept`)

Reply:
You are right, I was confused because `init_bridges_` has a `throw` line, but I believe this can be removed.

Reply:
Which can be removed? 
`init_bridges_` can throw exception
`discovered_topic_` cannot